libraries/chain/include/eosio/chain/webassembly/eos-vm-oc/config.hpp

```diff
@@ -16,11 +16,36 @@
 namespace eosio { namespace chain { namespace eosvmoc {
 
 struct config {
+   uint64_t get_cache_size() const { return cache_size; }
+   uint64_t get_threads() const { return threads; }
+
+   // empty optional means no rlimit
+   std::optional<rlim_t> get_cpu_limit() const { return !whitelisted ? cpu_limit : std::optional<rlim_t>{}; }
+   // empty optional means no rlimit
+   std::optional<rlim_t> get_vm_limit() const { return !whitelisted ? vm_limit : std::optional<rlim_t>{}; }
+
+   uint64_t get_stack_size_limit() const {
+      return !whitelisted
+                ? stack_size_limit
+                     ? *stack_size_limit
+                     : std::numeric_limits<uint64_t>::max()
+                : std::numeric_limits<uint64_t>::max();
+   }
+
+   size_t get_generated_code_size_limit() const {
+      return !whitelisted
+                ? generated_code_size_limit
+                     ? *generated_code_size_limit
+                     : std::numeric_limits<uint64_t>::max()
+                : std::numeric_limits<uint64_t>::max();
+   }
+
+   bool whitelisted = false;
```

Comment:
This really feels like we're dual purposing a struct that represents global static configuration with something that represents a per-compile input.

My first thought on how to resolve this commingling is to refactor out the subjective limits to their own struct
```cpp
struct subjective_compile_limits {
   std::optional<rlim_t>   cpu_limit {20u};
   //...
   std::optional<size_t>   generated_code_size_limit {16u*1024u*1024u};
};
```
Then make config be something like
```cpp
struct config {
   uint64_t cache_size = 1024u*1024u*1024u;
   //...
   subjective_compile_limits non_whitelisted_limits;
};
```
And then make `compile_wasm_message` not take a config but rather a limits struct. e.g. maybe
```cpp
struct compile_wasm_message {
   code_tuple code;
   eosvmoc::subjective_compile_limits limits;
};
```
Then when sending a `compile_wasm_message`, can decide how to populate `limits` based on whether it's whitelisted or not. (you maybe could also make `limits` an optional here)

Reply:
https://github.com/AntelopeIO/spring/pull/1010/commits/ffe173769af28b958715dab0836f33e627fd3a2d